### PR TITLE
feat: add support for router config generation to composition-go

### PR DESCRIPTION
**Pull request by fiam**
Also, use it to generate the router configuration when running
integration tests.

## Motivation and Context

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

File: composition-go/composition.go [hunk 1]
```diff
@@ -41,9 +50,60 @@ var (
...
+		req.Header.Add("Content-Type", "application/json")
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			return nil, fmt.Errorf("could not retrieve schema for subgraph %s: %w", subgraph.Name, err)
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode != http.StatusOK {
+			return nil, fmt.Errorf("could not retrieve schema for subgraph %s: unexpected status code %d", subgraph.Name, resp.StatusCode)
+		}
+		var sdlResult struct {
```
Comment (jensneuse) [inline composition-go/composition.go:73]: How about extracting this into a struct?

File: composition-go/composition.go [hunk 2]
```diff
@@ -41,9 +50,60 @@ var (
...
 )
 
+func updateSchemas(subgraphs []*Subgraph) ([]*Subgraph, error) {
+	updatedSubgraphs := make([]*Subgraph, 0, len(subgraphs))
+	for _, subgraph := range subgraphs {
+		if subgraph.Schema != "" {
+			updatedSubgraphs = append(updatedSubgraphs, subgraph)
+			continue
+		}
+		req, err := http.NewRequest("POST", subgraph.URL, strings.NewReader(sdlQuery))
```
Comment (jensneuse) [inline composition-go/composition.go:60]: How about moving the introspection of a subgraph into a separate function?

**Review by jensneuse — APPROVED** [review]

File: composition-go/composition.go (hunk 1 above)
Comment (fiam) [reply composition-go/composition.go]: Sounds reasonable, I've moved it to a type

File: composition-go/composition.go (hunk 2 above)
Comment (fiam) [reply composition-go/composition.go]: Good idea, that way we can also simplify the error messages a bit. I've refactor it, PTAL.

**Comment by fiam** [comment]
> Looks great except I miss comprehensive documentation of keeping in sync with composition library and using the go library to generate the config. Currently, it is spread around different folders. A README with some instructions would be great.

I've added a new section to the README that documents how to update the bundled code, PTAL.

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**